### PR TITLE
Remove dynamo suported check for Windows.

### PR DESCRIPTION
**Pull request by stellaraccident**
We've had developers using Dynamo via torch.compile and torch._dynamo.export for months on Windows via SHARK (by just deleting these lines locally). The Windows check is moved to just the inductor backend, since it is based on Triton not supporting Windows.

Progress on #90768

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Comment by Chillee** [comment]
Oh this is a good point. I think it mostly reflects that Triton does not work on Windows. We should move the check to the Inductor backend and not put it at the top level for Dynamo.

**Comment by stellaraccident** [comment]
> Oh this is a good point. I think it mostly reflects that Triton does not work on Windows. We should move the check to the Inductor backend and not put it at the top level for Dynamo.

Thanks. Would you like me to take a shot at this? I imagine you could probably do it faster than reviewing my attempt at updating code I don't frequently touch.

**Comment by stellaraccident** [comment]
Added the check to the dynamo backend. I imagine that there are also some tests that can have their Windows excludes removed with this?

**Review by jansel — CHANGES_REQUESTED** [review]
Looks like lints are failing, you should be able to fix with:
```
make setup_lint
lintrunner -a
```

**Comment by stellaraccident** [comment]
> Looks like lints are failing, you should be able to fix with:
> 
> ```
> make setup_lint
> lintrunner -a
> ```

Thanks. Applied the lint patch.

**Review by jansel — APPROVED** [review]

**Review by jansel — CHANGES_REQUESTED** [review]
@stellaraccident the test failure above looks real

**Comment by stellaraccident** [comment]
> @stellaraccident the test failure above looks real

Thanks - can add it to my list to triage. Does anyone know how to trigger these specific tests on presubmit?

**Comment by stellaraccident** [comment]
> Looks like this PR hasn't been updated in a while so we're going to go ahead and mark this as `Stale`. Feel free to remove the `Stale` label if you feel this was a mistake. If you are unable to remove the `Stale` label please contact a maintainer in order to do so. If you want the bot to never mark this PR stale again, add the `no-stale` label.`Stale` pull requests will automatically be closed after 30 days of inactivity.

This is still needed, afaict but I'd be open to handing it off: I'm not an expert enough on the pt windows side to get this all the way without some significant time spent.

**Comment by dan-garvey** [comment]
Can someone familiar please help me address the windows builder issues? I don't have any experience in the pytorch repository, but I'm constantly deleting the check for windows, and we're trying to bring up a windows ci that would also require deleting this line to function.

**Comment by nirvedhmeshram** [comment]
I think the reason the tests were failing was that the tests are skipped based on if dynamo is enabled which is the same is if the platform is not windows, so now with it being enabled new tests are being run on windows which is not the intent of this PR, so I am thinking to change the skip condition from "not dynamo" to "is WIndows" may be a solution, but I want to see what the CI does currently before going for that.

**Review by jansel — CHANGES_REQUESTED** [review]
Tests are failing and there are lints.  Does someone have a windows machine who can repro this?

**Comment by nirvedhmeshram** [comment]
> Tests are failing and there are lints. Does someone have a windows machine who can repro this?

I pushed a commit using the lint steps described previously, I was trying to get this to pass simply by looking at the CI but looks like I have to build locally and solve the issues, I do have a windows setup, so will try to fix on that, thanks.

**Comment by nirvedhmeshram** [comment]
@jansel I am building both on linux and windows and comparing why the `test_runtime_assert_inline_constraints_for_cond` test is passing on the former but not the later, what I have found is that, the `fn = self.pop()` here
https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/symbolic_convert.py#L1195
gives `UserFunctionVariable` for linux but `SkipFilesVariable` for windows, any thoughts why that would be?

**Comment by stellaraccident** [comment]
> @jansel I am building both on linux and windows and comparing why the `test_runtime_assert_inline_constraints_for_cond` test is passing on the former but not the later, what I have found is that, the `fn = self.pop()` here https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/symbolic_convert.py#L1195 gives `UserFunctionVariable` for linux but `SkipFilesVariable` for windows, any thoughts why that would be?

Last time I looked at this, there was a partial patch I found which fixed the path separator for a number of files related to that (and I think needed an exact string match). It was part of a bigger effort but that part looked relevant. It's been months, though, and I can't remember.

**Comment by jansel** [comment]
Possibly some of the logic in https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/skipfiles.py

Or https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/trace_rules.py

cc @yanboliang

**Comment by yanboliang** [comment]
From the error log:
```
torch._dynamo.exc.Unsupported: 'skip function cond in file C:\actions-runner\_work\pytorch\pytorch\build\win_tmp\build\torch\_higher_order_ops\cond.py'', skipped according skipfiles.SKIP_DIRS'
```
It looks like the forced inlined file ```torch._higher_order_ops.cond``` doesn't get inlined correctly. Can anyone run this unit tests with ```TORCH_LOGS="+dynamo"``` and share the output logs here?

**Comment by nirvedhmeshram** [comment]
> From the error log:
> 
> ```
> torch._dynamo.exc.Unsupported: 'skip function cond in file C:\actions-runner\_work\pytorch\pytorch\build\win_tmp\build\torch\_higher_order_ops\cond.py'', skipped according skipfiles.SKIP_DIRS'
> ```
> 
> It looks like the forced inlined file `torch._higher_order_ops.cond` doesn't get inlined correctly. Can anyone run this unit tests with `TORCH_LOGS="+dynamo"` and share output logs here?

Here is a gist of the output logs https://gist.github.com/nirvedhmeshram/69e1f8499e3c2a8341563df7e4ae6182

**Comment by yanboliang** [comment]
Can you run the the following code and share the output?
```
>>> import torch
>>> torch.cond.__code__
<code object cond at 0x7f44d43c5a50, file "/home/ybliang/local/pytorch/torch/_higher_order_ops/cond.py", line 36>
>>> from functorch.experimental.control_flow import cond
>>> cond.__code__
<code object cond at 0x7f44d43c5a50, file "/home/ybliang/local/pytorch/torch/_higher_order_ops/cond.py", line 36>
```

**Comment by nirvedhmeshram** [comment]
> Can you run the the following code and share the output?
> 
> ```
> >>> import torch
> >>> torch.cond.__code__
> <code object cond at 0x7f44d43c5a50, file "/home/ybliang/local/pytorch/torch/_higher_order_ops/cond.py", line 36>
> >>> from functorch.experimental.control_flow import cond
> >>> cond.__code__
> <code object cond at 0x7f44d43c5a50, file "/home/ybliang/local/pytorch/torch/_higher_order_ops/cond.py", line 36>
> ```

looks ok to me
```
>>> import torch
>>> torch.cond.__code__
<code object cond at 0x000001F241E78DB0, file "D:\shark_experiments\pytorch\torc
h\_higher_order_ops\cond.py", line 36>
>>> from functorch.experimental.control_flow import cond
>>> cond.__code__
<code object cond at 0x000001F241E78DB0, file "D:\shark_experiments\pytorch\torc
h\_higher_order_ops\cond.py", line 36>
```

**Comment by yanboliang** [comment]
@nirvedhmeshram This is weird, as we already put it in the force inline list:
https://github.com/pytorch/pytorch/blob/23b030a79caf88007eb7a17eaa14e2368eed8b06/torch/_dynamo/skipfiles.py#L173C12-L173C29

And it works well on other platform, not sure why it was wrapped as ```SkipFilesVariable``` on win, which is supposed to ```UserFunctionVariable```.

**Comment by nirvedhmeshram** [comment]
> @nirvedhmeshram This is weird, as we already put it in the force inline list: https://github.com/pytorch/pytorch/blob/23b030a79caf88007eb7a17eaa14e2368eed8b06/torch/_dynamo/skipfiles.py#L173C12-L173C29
> 
> And it works well on other platform, not sure why it was wrapped as `SkipFilesVariable` on win, which is supposed to `UserFunctionVariable`.

Its because of many places using "/" as seperator in file paths while windows need a different seperator so indeed the kind of fixes done here https://github.com/pytorch/pytorch/pull/109677/files are passing the tests for me. I need to apply the fixes properly and then will push up the fix, hopefully in a few hours.

**Comment by yanboliang** [comment]
> Its because of many places using "/" as seperator in file paths while windows need a different seperator so indeed the kind of fixes done here https://github.com/pytorch/pytorch/pull/109677/files are passing the tests for me. I need to apply the fixes properly and then will push up the fix, hopefully in a few hours.

That makes sense, but I'm curious why our CI doesn't capture this, since the skip/inline rules are used everywhere.

**Comment by nirvedhmeshram** [comment]
> make setup_lint

I think this is dynamo specific (at least the changes I am making are all in /torch/_dynamo) and dynamo was disabled on windows so it never hit these issues.

**Comment by nirvedhmeshram** [comment]
I had to disable `test_export_then_compile_tensor_ctor` test as it is using inductor which is not set up to find the cpp compiler as of now. Since this patch is trying to enable dynamo without trying to get inductor working, I hope that is okay. (Will ask for review after the CI passes.)

**Comment by monorimet** [comment]
Looking at the CI failure here,

```
Public FX API(s) ['torch.fx.passes.backends.cudagraphs.CudaGraphsSupport', 'torch.fx.passes.backends.cudagraphs.partition_cudagraphs', 'torch.fx.passes.infra.partitioner.CapabilityBasedPartitioner', 'torch.fx.passes.infra.partitioner.Partition'] introduced but not given a backwards-compatibility classification! Please decorate these API(s) with `@torch.fx._compatibility.compatibility` to specify BC guarantees.
```

can these decorators simply be added as suggested, or is there more to this failure that isn't being addressed in the logs?

**Comment by stellaraccident** [comment]
Closing because we think that #115969 addresses this.
